wasm/main.go

```diff
@@ -1,69 +1,720 @@
 				if err != nil {
-					log.Error(err.Error())
-					reject.Invoke("Failed")
+					log.Error("Error in getting response in GetEarning: ", err.Error())
 					return
 				}
 				defer resp.Body.Close()
 				respBuf, err := ioutil.ReadAll(resp.Body)
 				if err != nil {
-					log.Error(err.Error())
-					reject.Invoke("Failed")
+					log.Error("Error in reading Body in GetEarning: ", err.Error())
+					return
+				}
+				data := make(map[string]string)
+				err = json.Unmarshal(respBuf, &data)
+				if err != nil {
+					log.Error("Error in unmarshalling respBuf in GetEarning: ", err.Error())
 					return
 				}
-				log.Debug(string(respBuf))
-				resolve.Invoke(string(respBuf))
-			}()
 
+				var out Out
+				err = json.Unmarshal([]byte(data["val"]), &out)
+				if err != nil {
+					log.Error("Error in unmarshalling data in GetEarning: ", err.Error())
+					return
+				}
+				val, err := json.MarshalIndent(out.Data, "", " ")
+				if err != nil {
+					log.Error("Error in marshalling out in GetEarning: ", err.Error())
+					return
+				}
+				var netEarnings NetEarnings
+				err = json.Unmarshal(val, &netEarnings)
+				if err != nil {
+					log.Error("Error in Unmarshalling Net Earnings in GetEarning: ", err.Error())
+					return
+				}
+
+				jsDoc := js.Global().Get("document")
+				if !jsDoc.Truthy() {
+					log.Error("Unable to get document object in GetEarning")
+					return
+				}
+				OutputArea := jsDoc.Call("getElementById", "DevicesDropDown")
+				if !OutputArea.Truthy() {
+					log.Error("Unable to get output area in GetEarning")
+					return
+				}
+				value := fmt.Sprintf("%s", OutputArea.Get("value"))
+				var earned, download, served string
+				if value == "ALL DEVICES" {
+					earned = fmt.Sprintf("%.5f %s", netEarnings.DeviceTotal.Earned, "SWRM")
+					download = fmt.Sprintf("%.0f %s", (netEarnings.DeviceTotal.Download)/1048576, "MB")
+					served = fmt.Sprintf("%.2f %s", (netEarnings.DeviceTotal.Served)/1048576, "MB")
+				} else if value != "" {
+					earned = fmt.Sprintf("%.5f %s", netEarnings.Data[value][0].Earned, "SWRM")
+					download = fmt.Sprintf("%.2f %s", (netEarnings.Data[value][0].Download)/1048576, "MB")
+					served = fmt.Sprintf("%.2f %s", (netEarnings.Data[value][0].Served)/1048576, "MB")
+				}
+				SetDisplay("EarnedCycle", "innerHTML", earned)
+				SetDisplay("DownloadedCycle", "innerHTML", download)
+				SetDisplay("ServedCycle", "innerHTML", served)
+
+				log.Debug("Sending details to CreateGraph from GetEarning")
+				resolve.Invoke(data["val"])
+			}()
 			return nil
 		})
-
 		promiseConstructor := js.Global().Get("Promise")
 		return promiseConstructor.New(handler)
 		return nil
 	})
 	return jsonFunc
 }
 
+func GetUptime() js.Func {
+	jsonFunc := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		go func() {
+			Start := time.Unix(StartTime, 0)
+			elapsed := time.Since(Start)
+			SetDisplay("Time", "innerHTML", fmt.Sprintf("%s", durafmt.Parse(elapsed).LimitFirstN(1)))
```

Comment:
Dont show this until `StartTime` is assigned with `SessionStartTime`

Reply:
Okay